code/game/objects/items/binoculars.dm

```diff
@@ -293,19 +305,18 @@
 
 ///Sets or unsets the binocs linked mortar.
 /obj/item/binoculars/tactical/proc/set_mortar(mortar)
-	if(linked_mortar)
-		UnregisterSignal(linked_mortar, COMSIG_PARENT_QDELETING)
-	if(linked_mortar == mortar)
-		linked_mortar = null
+	if(mortar in linked_mortars)
+		UnregisterSignal(mortar, COMSIG_PARENT_QDELETING)
+		linked_mortars -= mortar
 		return FALSE
-	linked_mortar = mortar
-	RegisterSignal(linked_mortar, COMSIG_PARENT_QDELETING, .proc/clean_refs)
+	linked_mortars += mortar
+	RegisterSignal(mortar, COMSIG_PARENT_QDELETING, .proc/clean_refs)
 	return TRUE
 
 ///Proc called when linked_mortar is deleted.
-/obj/item/binoculars/tactical/proc/clean_refs()
+/obj/item/binoculars/tactical/proc/clean_refs(datum/source)
```

Comment:
That is probably messing with index.

Reply:
Its fine because if you try to send coords to that index it simply defaults back to 1 if theres an existent mortar

Reply:
although i should make the defaulting a proc hold on

Reply:
done